sabnzbd/nzbstuff.py

```diff
@@ -187,65 +187,45 @@ def reset_try_list(self):
     def get_article(self, server: Server, servers: List[Server]):
         """Return article when appropriate for specified server"""
         log = sabnzbd.LOG_ALL
-        if not self.fetcher and not self.server_in_try_list(server):
+
+        if self.fetcher or self.server_in_try_list(server):
             if log:
-                logging.debug("Article %s | Server: %s | in second if", self.article, server.host)
-                # Is the current selected server of the same priority as this article?
-                logging.debug(
-                    "Article %s | Server: %s | Article priority: %s", self.article, server.host, self.fetcher_priority
-                )
-                logging.debug(
-                    "Article %s | Server: %s | Server priority: %s", self.article, server.host, server.priority
-                )
-            if server.priority == self.fetcher_priority:
-                self.fetcher = server
-                self.tries += 1
-                if log:
-                    logging.debug("Article %s | Server: %s | same priority, use it", self.article, server.host)
-                    logging.debug("Article %s | Server: %s | Article-try: %s", self.article, server.host, self.tries)
-                return self
-            else:
+                logging.debug("Article %s | Server: %s | Returning None", self.article, server.host)
+            return None
+
+        if log:
```

Comment:
Looks good. Let's get rid of all the `if log`'s. They make the code very hard to read.